panther_manager/include/panther_manager/plugins/shutdown_hosts_node.hpp

```diff
@@ -0,0 +1,150 @@
+#ifndef PANTHER_MANAGER_SHUTDOWN_HOSTS_NODE_HPP_
+#define PANTHER_MANAGER_SHUTDOWN_HOSTS_NODE_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <set>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <behaviortree_cpp/action_node.h>
+#include <behaviortree_cpp/basic_types.h>
+#include <behaviortree_cpp/tree_node.h>
+
+#include <ros/ros.h>
+
+#include <panther_manager/plugins/shutdown_host.hpp>
+
+namespace panther_manager
+{
+
+class ShutdownHosts : public BT::StatefulActionNode
+{
+public:
+  explicit ShutdownHosts(const std::string & name, const BT::NodeConfig & conf)
+  : BT::StatefulActionNode(name, conf)
+  {
+    node_name_ = ros::this_node::getName();
+  }
+
+  virtual ~ShutdownHosts() = default;
+
+  // method to be implemented by user
+  virtual void update_hosts(std::vector<std::shared_ptr<ShutdownHost>> & hosts) = 0;
+
+  // method that can be overriden by user
+  virtual BT::NodeStatus post_process()
+  {
+    // return success only when all hosts succeded
+    if (failed_hosts_.size() == 0) return BT::NodeStatus::SUCCESS;
+    return BT::NodeStatus::FAILURE;
```

Comment:
```suggestion
    if (failed_hosts_.size() == 0) {
      return BT::NodeStatus::SUCCESS;
    }
```
Missing brackets make the code look like `FAILURE` is being conditioned with above `if`